libtracepoint-control-cpp/include/tracepoint/TracepointSession.h

```diff
@@ -1084,8 +1170,21 @@ namespace tracepoint_control
             uint32_t bufferIndex,
             RecordFn&& recordFn) noexcept(noexcept(recordFn(nullptr, 0, 0)));
 
+        _Success_(return == 0) int
+        SetTracepointEnableState(
+            TracepointInfoImpl& tpi,
+            bool enabled) noexcept;
+
+        _Success_(return == 0) int
+        AddTracepoint(
+            tracepoint_decode::PerfEventMetadata const& metadata,
+            TracepointEnableState enableState) noexcept(false);
+
```

Comment:
Note: Implementation detail - refactor these methods out of the EnableTracepointImpl method.